actionpack/lib/action_controller/metal/helpers.rb

```diff
@@ -74,10 +74,9 @@ def helper_attr(*attrs)
 
       # Provides a proxy to access helper methods from outside the view.
       def helpers
-        @helper_proxy ||= begin
-          proxy = ActionView::Base.empty
-          proxy.config = config.inheritable_copy
-          proxy.extend(_helpers)
+        @helpers_proxy ||= begin
+          helpers = renderer.helpers
+          helpers.extend(_helpers)
```

Comment:
Do we need to  `extend(_helpers)` ?

Reply:
yes, before doing the `extend(_helpers)` these [tests](https://github.com/rails/rails/blob/master/actionpack/test/controller/api/with_helpers_test.rb) since they `ApiWithHelper` wouldn't be included in the controller `helpers` instance method. 

One think I tried was to do something like in the controller instance method:

```` ruby
def helpers
  @_helpers_proxy ||= view_context.extend(_helpers)
end
```

but at that point the `_helpers` is empty.